src/Google.Cloud.Functions.Invoker.Testing/FunctionTestBase.cs

```diff
@@ -0,0 +1,124 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using CloudNative.CloudEvents;
+using Microsoft.AspNetCore.Authorization.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Mime;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Functions.Invoker.Testing
+{
+    /// <summary>
+    /// Base class for test for functions. This class is designed
+    /// for use in test frameworks that dispose of test classes
+    /// automatically. (This includes NUnit and xUnit.) This class
+    /// creates a <see cref="FunctionTestServer{TFunction}"/> on construction,
+    /// and disposes of the server when the instance of this class is disposed.
+    /// Additionally, convenience methods to make requests to the server and
+    /// access log entries are provided.
+    /// </summary>
+    public abstract class FunctionTestBase<TFunction> : IDisposable
+    {
+        private static readonly ICloudEventFormatter s_eventFormatter = new JsonEventFormatter();
+        private static readonly ContentType s_applicationJsonContentType = new ContentType("application/json");
+
+        /// <summary>
+        /// The server which will invoke <typeparamref name="TFunction"/>
+        /// when a request is made.
+        /// </summary>
+        public FunctionTestServer<TFunction> Server { get; }
+
+        /// <summary>
+        /// Constructs a new instance using the specified server. Note that
+        /// this effectively takes ownership of the given server, which will be
+        /// disposed when the FunctionTestBase instance is disposed.
+        /// </summary>
+        /// <param name="server">The function test server.</param>
+        protected FunctionTestBase(FunctionTestServer<TFunction> server)
+        {
+            Server = server;
+        }
+
+        /// <summary>
+        /// Constructs a new instance using a new server constructed with default settings.
+        /// </summary>
+        protected FunctionTestBase() : this(new FunctionTestServer<TFunction>())
+        {
+        }
+
+        /// <summary>
+        /// Executes a simple HTTP GET request against an arbitrary URI, and returns
+        /// the text of the response after validating that it indicates success. For
+        /// more complex scenarios, use <see cref="ExecuteHttpRequestAsync(HttpRequestMessage, Action{HttpResponseMessage})"/>.
+        /// </summary>
+        public async Task<string> ExecuteHttpGetRequestAsync()
```

Comment:
Consider an optional uri parameter. There might be functions that validate the calling uri I suppose.